applications/crossbar/src/crossbar_maintenance.erl

```diff
@@ -867,6 +869,46 @@ init_apps(AppsPath, AppUrl) ->
     InitApp = fun(App) -> init_app(App, AppUrl) end,
     lists:foreach(InitApp, Apps).
 
+-spec init_app(file:filename_all()) -> 'ok'.
+init_app(AppPath) ->
+    init_app(AppPath, 'undefined').
+
+-spec init_app(file:filename_all(), kz_term:api_binary()) -> 'ok'.
+init_app(AppPath, AppUrl) ->
+    Keys = [<<"api_url">>],
+    io:format("trying to init app from ~s~n", [AppPath]),
+    maybe_update_app(AppPath, AppUrl, Keys).
+
+-spec refresh_apps(file:name()) -> 'ok'.
+refresh_apps(AppsPath) ->
+    refresh_apps(AppsPath, 'undefined').
+
+-spec refresh_apps(file:name(), kz_term:api_binary()) -> 'ok'.
+refresh_apps(AppsPath, AppUrl) ->
+    Apps = find_apps(AppsPath),
+    InitApp = fun(App) -> refresh_app(App, AppUrl) end,
+    lists:foreach(InitApp, Apps).
+
+-spec refresh_app(file:filename_all()) -> 'ok'.
+refresh_app(AppPath) ->
+    refresh_app(AppPath, 'undefined').
+
+-spec refresh_app(file:filename_all(), kz_term:api_binary()) -> 'ok'.
+refresh_app(AppPath, AppUrl) ->
+    Keys = [<<"name">>
```

Comment:
Should we pull this from kapps_config? Would make it easier to extend to other keys vs code changes.